core/state_transition.go

```diff
@@ -253,7 +253,8 @@ func (self *StateTransition) transitionDb() (ret []byte, usedGas *big.Int, err e
 		err = nil
 	}
 
-	if vm.Debug {
+	logs := vmenv.StructLogs()
+	if len(logs) > 0 {
```

Comment:
This change makes it print all struct logs to stderr. Why did you add this?
If users (e.g. etherchain) request struct logs through the API, their logs grow very large.